src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,897 @@
+            System.out.println("Noted. I've removed this task:");
+            task.print();
+            System.out.println("Now you have " + len + " tasks in the list.");
+            System.out.println(horizontalLine);
+        }
+
+         /**
+         * Returns void, just prints the latest task marked
+         * 
+         * @param task the task that was marked
+         */
+        void printTaskMarked(Task task) {
+            System.out.println(horizontalLine);
+            System.out.println("Nice! I've marked this task as done:");
+            task.print();
+            System.out.println(horizontalLine);
+        }
+
+         /**
+         * Returns void, just prints the latest task unmarked
+         * 
+         * @param task the task that was unmarked
+         */
+        void printTaskUnmarked(Task task) {
+            System.out.println(horizontalLine);
+            System.out.println("OK, I've marked this task as not done yet:");
+            task.print();
+            System.out.println(horizontalLine);
+        }
+
+         /**
+         * Returns void, just prints error messages
+         * 
+         * @param str the error message to print
+         */
+        void printError(String str) {
+            System.out.println(horizontalLine);
+            System.out.println(str);
+            System.out.println(horizontalLine);
+        }
+
+        String openingText;
+        String closingText;
+        String horizontalLine;
+    }
+
+    public static class Parser {
+
+        /**
+         * Returns a Parser object for parsing user commands
+         */
+        public Parser() {
+            userInput = "";
+            this.ptr = 0;
+        }
+
+        /**
+         * Returns a Parser object for parsing user commands
+         * 
+         * @param userInput the entire command the user has sent
+         * @return Parser object
+         */
+        public Parser(String userInput) {
+            this.userInput = userInput;
+            this.ptr = 0;
+        }
+
+
+        /**
+         * Returns void, just resets the parser state to match new
+         * user command that was inputted
+         * 
+         * @param newUserInput the new command the user has inputted
+         */
+        public void readInput(String newUserInput) {
+            this.userInput = newUserInput;
+            this.ptr = 0;
+        }
+
+        /**
+         * Returns a substring of this.userInput starting from ptr
+         * until the first instance of the escapeCharacter that
+         * appears past ptr.
+         * 
+         * @param escapeCharacter the escape character that determines
+         * the end of the substring we should return
+         * @return the substring from ptr to escapeCharacter (exclusive)
+         */
+        public String getArgument(char escapeCharacter) {
+            int len = userInput.length();
+            int startIndex = ptr;
+
+            while(ptr < len) {
```

Comment:
The recommended Java coding standard requires a space after reserved keywords such as `while`. But you'll probably figure this out after adding checkstyle in Week 4, so you can fix it then!